src/coreclr/jit/morph.cpp

```diff
@@ -12737,6 +12791,17 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                 op2  = tree->AsOp()->gtOp2;
             }
 
+            if (varTypeIsIntegralOrI(tree->TypeGet()) && tree->OperIs(GT_AND, GT_OR, GT_XOR))
+            {
+                tree = fgMorphOpCast(tree->AsOp());
+
+                // fgMorphOpCast may return a new tree
+                oper = tree->OperGet();
+                typ  = tree->TypeGet();
+                op1  = tree->AsOp()->gtOp1;
+                op2  = tree->AsOp()->gtOp2;
```

Comment:
```suggestion
                op1  = tree->AsOp()->gtGetOp1();
                op2  = tree->AsOp()->gtGetOp2();
```

Reply:
We prefer to use getters whenever reasonable.

Reply:
cdec019 and 838b35b